tsl/src/fdw/scan_exec.c

```diff
@@ -152,6 +152,24 @@ create_data_fetcher(ScanState *ss, TsFdwScanState *fsstate)
 		}
 	}
 
+	/*
+	 * Row-by-row fetcher uses COPY statement that don't work with prepared
+	 * statements. If this plan is parameterized, this means we'll have to
+	 * revert to cursor fetcher.
+	 */
+	if (num_params > 0 && fsstate->planned_fetcher_type == RowByRowFetcherType)
+	{
+		if (ts_guc_remote_data_fetcher == AutoFetcherType)
+		{
+			fsstate->planned_fetcher_type = CursorFetcherType;
+		}
+		else
+		{
+			ereport(ERROR,
+					(errmsg("cannot use row-by-row fetcher because the plan is parameterized")));
```

Comment:
I think it would be a good idea to elaborate on this error message and give some information why you print it and provide hints on how to fix the problem. In this case it is because the fetcher type is fixed and cannot be changed, so might be good to provide a hint suggesting to set the planner fetcher type to cursor manually or use the auto fetcher type.

Reply:
Added a hint about the GUC.